decidim-elections/app/forms/decidim/votings/admin/ballot_style_form.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Votings
+    module Admin
+      # A form to create/edit a ballot style
+      class BallotStyleForm < Form
+        attribute :title, String
+        attribute :code, String
+        attribute :question_ids, Array[Integer]
+
+        validates :code, presence: true
+        validate :code_uniqueness
+
+        private
+
+        def code_uniqueness
+          return unless voting_ballot_styles
+                        .where(code: code)
+                        .where.not(id: context[:ballot_style_id])
+                        .any?
```

Comment:
💬  you can use `exists?` instead of `any?`, because `any?` will load the relation in memory, while `exists?` will only retrieve a boolean result from the database. 

Anyway, I'd remove all the uniqueness validations like this when they can be replaced with a unique index. It does less queries to the database and prevent race conditions. In this case, it is possible with these steps:
* Remove this validation (and the `voting_ballot_styles` method).
* Remove the unique validation from the `BallotStyle` model.
* Add a unique index for the `voting_id` and `code` columns in the `ballot_styles` table.
* In the `CreateBallotStyle` and `UpdateBallotStyle` handle the `ActiveRecord::RecordNotUnique` when saving the `ballot_style` object, doing the `errors.add(:code, :taken)` and broadcasting an `:invalid` code.

Reply:
https://github.com/decidim/decidim/pull/7779/commits/92f64a294bb5507090486a9b351298b18ea43069